transaction-service/src/test/kotlin/ua/com/foxminded/restClient/kafka/KafkaConsumerServiceTest.kt

```diff
@@ -1,25 +1,63 @@
 package ua.com.foxminded.restClient.kafka
 
+import java.util.concurrent.CountDownLatch
+import kotlin.test.assertEquals
 import org.junit.jupiter.api.Test
+import org.mockito.Mockito
 import org.springframework.beans.factory.annotation.Autowired
-import org.springframework.boot.test.autoconfigure.web.reactive.AutoConfigureWebTestClient
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.beans.factory.annotation.Value
 import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.kafka.core.reactive.ReactiveKafkaConsumerTemplate
 import org.springframework.kafka.core.reactive.ReactiveKafkaProducerTemplate
+import org.springframework.kafka.test.context.EmbeddedKafka
 import proto.ProtoMessage
+import ua.com.foxminded.restClient.config.DBTestConfig
+import ua.com.foxminded.restClient.service.RateService
 import utils.Transactions
 
 @SpringBootTest
-@AutoConfigureWebTestClient
-class KafkaConsumerServiceTest {
+@EmbeddedKafka(topics = ["t1", "t2"], partitions = 1)
+class KafkaConsumerServiceTest : DBTestConfig() {
 
+    @Value(value = "\${kafka.consumer.topics}")
+    private lateinit var consumerTopics: List<String>
+
+    @MockBean
+    private lateinit var rateService: RateService
 
     @Autowired
     private lateinit var kafkaProducerTemplate: ReactiveKafkaProducerTemplate<String, ByteArray>
 
-    //IT isn't finished test. Don't review
+    @Autowired
+    @Qualifier("reactiveKafkaTestConsumerTemplate")
+    private lateinit var kafkaConsumerTemplate: ReactiveKafkaConsumerTemplate<String, ByteArray>
+
     @Test
     fun `findTransactionsByPersonIdAndTime should return ListResponse when input correct`() {
         val request: ProtoMessage.FindTransactionsByPersonIdAndTimeRequest = Transactions.transactionRequest
-        kafkaProducerTemplate.send("transactions",request.toByteArray()).subscribe()
+        val messageList: MutableList<ProtoMessage.FindTransactionsByPersonIdAndTimeListResponse> = mutableListOf()
+        val latch = CountDownLatch(1)
+        val expectedSize = 2
+        kafkaConsumerTemplate.receiveAutoAck()
+            .handle { obj, sink ->
+                val byteArray = obj.value() as ByteArray
+                when (String(byteArray)) {
+                    "Hello" -> {}
```

Comment:
how there can be hello?

Reply:
OMG, I can't explain it. I thought that it was some message for test when zookeeper and brokers up. But now I changed topics and I don't see Hello message.
May be it was in some topics, when i try to test it firstly.